src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ICastableHelpers.cs

```diff
@@ -22,5 +23,30 @@ internal static RuntimeType GetImplType(ICastable castable, RuntimeType interfac
         {
             return castable.GetImplType(new RuntimeTypeHandle(interfaceType)).GetRuntimeType();
         }
+
+        [Diagnostics.StackTraceHidden]
+        internal static RuntimeType? GetInterfaceImplementation(ICastableObject castableObject, RuntimeType interfaceType, bool throwIfNotFound)
+        {
+            RuntimeTypeHandle handle = castableObject.GetInterfaceImplementation(new RuntimeTypeHandle(interfaceType), throwIfNotFound);
+            if (handle.Equals(default))
+            {
+                if (throwIfNotFound)
+                    throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, castableObject.GetType().ToString(), interfaceType.ToString()));
```

Comment:
I do not think you need to call `.ToString()` for the formatting strings. `SR.Format` will do it for you.